supabase/migrations/20251021110000_avatar_url_and_storage.sql

```
@@ -0,0 +1,79 @@
+-- storage.foldername(name) splits that path into an array of folder
+-- segments, so (storage.foldername(name))[1] is that user-id segment.
+-- The filename portion is intentionally unconstrained by these policies, so
+-- the upload implementation is free to use a fixed name (e.g. always
+-- "avatar.jpg", overwritten on each upload) or a unique name per upload --
+-- whichever convention the Dart upload code adopts, ownership is enforced by
+-- the same first-path-segment check either way.
+
+-- Anyone (including anonymous/unauthenticated requests) can read avatar
+-- objects, since the bucket is public and avatars must render for any
+-- viewer in the app.
+DROP POLICY IF EXISTS "Public can view avatars" ON storage.objects;
```

SQLFluff (4.3.0)

[error] Do not use special characters in identifiers.

(RF05)